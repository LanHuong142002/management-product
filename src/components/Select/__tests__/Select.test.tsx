import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { MOCK_TYPE_API } from '@constants';
import { Select } from '@components';

describe('Testing Select component', () => {
  const handleChange = jest.fn();

  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    valueSelected: 'type',
    name: 'type',
    options: MOCK_TYPE_API,
    onChange: handleChange,
  };

  it('Should render Select component', () => {
    const { container } = render(<Select {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should render without title and optionAll', () => {
    render(<Select {...defaultProps} title='Full Name' />);

    const select = screen.getByTestId('select-box');

    expect(select).toMatchSnapshot();
  });

  it('should render select with optionAll correctly', () => {
    render(<Select {...defaultProps} optionAll={true} />);

    const select = screen.getByTestId('select-box');
    const optionAll = screen.getByDisplayValue('All');

    expect(select).toBeInTheDocument();
    expect(optionAll).toBeInTheDocument();
  });

  it('should call onChange when an option is selected', () => {
    const { getByDisplayValue } = render(<Select {...defaultProps} />);
    const select = getByDisplayValue('Bracelet');

    fireEvent.change(select, { target: { value: 'e76918cc-89fe-4329-99e5-63b939df3306' } });

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenCalledWith(expect.any(Object));
  });
});
