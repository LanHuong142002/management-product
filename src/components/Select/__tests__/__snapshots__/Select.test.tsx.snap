// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Select component Should render Select component 1`] = `
<div>
  <div
    data-testid="select-box"
  >
    <select
      class="select-wrapper"
      name="type"
    >
      <option
        class="select-item"
        value="f678be1f-ce92-4483-82b9-9fb06e88b196"
      >
        Bracelet
      </option>
      <option
        class="select-item"
        value="e76918cc-89fe-4329-99e5-63b939df3306"
      >
        Ring
      </option>
      <option
        class="select-item"
        value="39919010-766d-4672-81b3-f60cfdadfbd9"
      >
        Watch
      </option>
    </select>
  </div>
</div>
`;

exports[`Testing Select component should render without title and optionAll 1`] = `
<div
  class="select-box"
  data-testid="select-box"
>
  <label
    class="title-select"
  >
    Full Name
  </label>
  <select
    class="select-wrapper"
    name="type"
  >
    <option
      class="select-item"
      value="f678be1f-ce92-4483-82b9-9fb06e88b196"
    >
      Bracelet
    </option>
    <option
      class="select-item"
      value="e76918cc-89fe-4329-99e5-63b939df3306"
    >
      Ring
    </option>
    <option
      class="select-item"
      value="39919010-766d-4672-81b3-f60cfdadfbd9"
    >
      Watch
    </option>
  </select>
</div>
`;
